src/components/AuthGuard/AuthGuard.tsx

```diff
@@ -33,11 +34,17 @@ const AuthGuard: FC = () => {
         // Get user
         if (!user || (user?.uuid !== oidcUser.profile.sub && !auth.isLoading)) {
           try {
-            await dispatch(getUserThunk());
+            //await dispatch(getUserThunk());
 
-            if (!user || user.ad_groups.length === 0) {
+            // Get user payload to check AD groups
+            const userDispatch = await dispatch(getUserThunk());
+            const userPayload = userDispatch.payload as IUser;
+
+            //if (!user || user.ad_groups.length === 0) {
+            if (userDispatch.type.includes("fulfilled") && userPayload.ad_groups.length === 0) {
+              // Do not redirect user to the path where tried to access
               localStorage.removeItem(INITIAL_PATH);
-              return;
+              getApiToken();
             }
```

Comment:
In some reason, this `dispatch` does not affect to `user`, so every time when `user` is called it is `null`.

This is why redirecting will not work when returning to the app after signing in.

This change works, but it should use `useAppSelector(selectUser)` and not fetch the information using payload.